src/ray/core_worker/core_worker.cc

```diff
@@ -157,22 +160,29 @@ CoreWorker::CoreWorker(const WorkerType worker_type, const Language language,
     std::vector<std::string> empty_descriptor;
     std::unordered_map<std::string, double> empty_resources;
     const TaskID task_id = TaskID::ForDriverTask(worker_context_.GetCurrentJobID());
-    builder.SetCommonTaskSpec(task_id, language_, empty_descriptor,
-                              worker_context_.GetCurrentJobID(),
-                              TaskID::ComputeDriverTaskId(worker_context_.GetWorkerID()),
-                              0, GetCallerId(), 0, empty_resources, empty_resources);
+    builder.SetCommonTaskSpec(
+        task_id, language_, empty_descriptor, worker_context_.GetCurrentJobID(),
+        TaskID::ComputeDriverTaskId(worker_context_.GetWorkerID()), 0, GetCallerId(), 0,
+        false, empty_resources, empty_resources);
 
     std::shared_ptr<gcs::TaskTableData> data = std::make_shared<gcs::TaskTableData>();
     data->mutable_task()->mutable_task_spec()->CopyFrom(builder.Build().GetMessage());
     RAY_CHECK_OK(gcs_client_.raylet_task_table().Add(job_id, task_id, data, nullptr));
     SetCurrentTaskId(task_id);
   }
 
-  // TODO(edoakes): why don't we just share the memory store provider?
   direct_actor_submitter_ = std::unique_ptr<CoreWorkerDirectActorTaskSubmitter>(
-      new CoreWorkerDirectActorTaskSubmitter(
-          io_service_, std::unique_ptr<CoreWorkerMemoryStoreProvider>(
-                           new CoreWorkerMemoryStoreProvider(memory_store_))));
+      new CoreWorkerDirectActorTaskSubmitter(*client_call_manager_,
+                                             memory_store_provider_));
+
+  direct_task_submitter_ =
+      std::unique_ptr<CoreWorkerDirectTaskSubmitter>(new CoreWorkerDirectTaskSubmitter(
+          *raylet_client_,
+          [this](WorkerAddress addr) {
+            return std::shared_ptr<rpc::CoreWorkerClient>(new rpc::CoreWorkerClient(
+                addr.first, addr.second, *client_call_manager_));
```

Comment:
why is this necessary for `direct_task_submitter` but not `direct_actor_submitter`?

Reply:
This is because the direct actor submitter doesn't need to request any scheduling decisions from the Raylet.